src/istio/authn/context.proto

```diff
@@ -51,6 +51,22 @@ message X509Payload {
   // Identity carries by X509 certification. This is extracted from SAN field
   // for Istio provided certificate.
   string user = 1;
+
+  // The trust domain corresponds to the trust root of a system.
+  // Refer to
+  // [SPIFFE-ID](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain)
+  string trust_domain = 2;
+
+  // The trust domain aliases represent the aliases of `trust_domain`.
+  // For example, if we have
+  // ```yaml
+  // trustDomain: td1
+  // trustDomainAliases: [“td2”, "td3"]
+  // ```
+  // Any service with the identity `td1/ns/foo/sa/a-service-account`,
+  // `td2/ns/foo/sa/a-service-account`, or `td3/ns/foo/sa/a-service-account`
+  // will be treated the same in the Istio mesh.
+  repeated string trust_domain_aliases = 3;
```

Comment:
Why are we using aliases instead of defining multiple trust domains (i.e. `repeated string trust_domains`) above?

Reply:
Actually this is a good point. The change currently follows what we have in MeshConfig API. But since authentication filter does not need to distinguish trust_domain vs. trust_domain_alias, they can be collapsed into a single trust_domains field. This also simplifies implementation.

Reply:
I think we should keep it as it is to align with MeshConfig. Later on, we need to distinguish between "cluster.local" as the local trust domain -> "cluster.local" as alias, and "some-td" as local trust domain -> "cluster.local" as the local trust domain. Also, this will help later with removing the aliases only when the users are done with the migration.

Reply:
Per offline discussion, I merged trust domain and trust domain aliases field to one. PTAL. Thanks!